src/platform_impl/macos/event.rs

```diff
@@ -25,216 +41,283 @@ pub(crate) enum EventProxy {
-        0x35 => VirtualKeyCode::Escape,
-        0x36 => VirtualKeyCode::RWin,
-        0x37 => VirtualKeyCode::LWin,
-        0x38 => VirtualKeyCode::LShift,
-        //0x39 => Caps lock,
-        0x3a => VirtualKeyCode::LAlt,
-        0x3b => VirtualKeyCode::LControl,
-        0x3c => VirtualKeyCode::RShift,
-        0x3d => VirtualKeyCode::RAlt,
-        0x3e => VirtualKeyCode::RControl,
-        //0x3f => Fn key,
-        0x40 => VirtualKeyCode::F17,
-        0x41 => VirtualKeyCode::NumpadDecimal,
-        //0x42 -> unkown,
-        0x43 => VirtualKeyCode::NumpadMultiply,
-        //0x44 => unkown,
-        0x45 => VirtualKeyCode::NumpadAdd,
-        //0x46 => unkown,
-        0x47 => VirtualKeyCode::Numlock,
-        //0x48 => KeypadClear,
-        0x49 => VirtualKeyCode::VolumeUp,
-        0x4a => VirtualKeyCode::VolumeDown,
-        0x4b => VirtualKeyCode::NumpadDivide,
-        0x4c => VirtualKeyCode::NumpadEnter,
-        //0x4d => unkown,
-        0x4e => VirtualKeyCode::NumpadSubtract,
-        0x4f => VirtualKeyCode::F18,
-        0x50 => VirtualKeyCode::F19,
-        0x51 => VirtualKeyCode::NumpadEquals,
-        0x52 => VirtualKeyCode::Numpad0,
-        0x53 => VirtualKeyCode::Numpad1,
-        0x54 => VirtualKeyCode::Numpad2,
-        0x55 => VirtualKeyCode::Numpad3,
-        0x56 => VirtualKeyCode::Numpad4,
-        0x57 => VirtualKeyCode::Numpad5,
-        0x58 => VirtualKeyCode::Numpad6,
-        0x59 => VirtualKeyCode::Numpad7,
-        0x5a => VirtualKeyCode::F20,
-        0x5b => VirtualKeyCode::Numpad8,
-        0x5c => VirtualKeyCode::Numpad9,
-        0x5d => VirtualKeyCode::Yen,
-        //0x5e => JIS Ro,
-        //0x5f => unkown,
-        0x60 => VirtualKeyCode::F5,
-        0x61 => VirtualKeyCode::F6,
-        0x62 => VirtualKeyCode::F7,
-        0x63 => VirtualKeyCode::F3,
-        0x64 => VirtualKeyCode::F8,
-        0x65 => VirtualKeyCode::F9,
-        //0x66 => JIS Eisuu (macOS),
-        0x67 => VirtualKeyCode::F11,
-        //0x68 => JIS Kanna (macOS),
-        0x69 => VirtualKeyCode::F13,
-        0x6a => VirtualKeyCode::F16,
-        0x6b => VirtualKeyCode::F14,
-        //0x6c => unkown,
-        0x6d => VirtualKeyCode::F10,
-        //0x6e => unkown,
-        0x6f => VirtualKeyCode::F12,
-        //0x70 => unkown,
-        0x71 => VirtualKeyCode::F15,
-        0x72 => VirtualKeyCode::Insert,
-        0x73 => VirtualKeyCode::Home,
-        0x74 => VirtualKeyCode::PageUp,
-        0x75 => VirtualKeyCode::Delete,
-        0x76 => VirtualKeyCode::F4,
-        0x77 => VirtualKeyCode::End,
-        0x78 => VirtualKeyCode::F2,
-        0x79 => VirtualKeyCode::PageDown,
-        0x7a => VirtualKeyCode::F1,
-        0x7b => VirtualKeyCode::Left,
-        0x7c => VirtualKeyCode::Right,
-        0x7d => VirtualKeyCode::Down,
-        0x7e => VirtualKeyCode::Up,
-        //0x7f =>  unkown,
-        0xa => VirtualKeyCode::Caret,
-        _ => return None,
-    })
+pub fn code_to_location(code: KeyCode) -> KeyLocation {
```

Comment:
Be careful that if you add this change, it should go in a separate commmit that's a fixup for the API changes commit.
✅

Reply:
> (since it's also used e.g. in the Wayland backend).

There's only really a similar function in the Android backend. The X11 and Wayland backends do share a function called `keysym_to_location`, however.

Reply:
I think this should be public regardless, because it might be a functionality that dependents wants to use. I think it would probably be best to add this as a method of `KeyCode` so people can write `code.get_location()`

Reply:
@madsmtm @maroider is it okay if I add this `get_location` function as a public method on `KeyCode`? I mean something like:

```rust
impl KeyCode {
   pub fn get_location(&self) -> KeyLocation {
       match self {
           // ...
       }
   }
}
```